src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1933,9 +1933,8 @@ public static SqlConnection OpenSqlConnection(ConnectionInfo connInfo, string fe
                     "Failed opening a SqlConnection: error:{0} inner:{1} stacktrace:{2}",
                     ex.Message, ex.InnerException != null ? ex.InnerException.Message : string.Empty, ex.StackTrace);
                 Logger.Write(TraceEventType.Error, error);
+                throw;
             }
-
-            return null;

```

Comment:
There wasn't any null handling after calling this method, so it made more sense to just throw the exception above.